$NetBSD: patch-aa,v 1.3 2008/01/29 03:45:36 bjs Exp $

Coverity #743/744: Returned without freeing storage bufp/savp
 
If either bufp or savp failed to malloc, we returned without freeing 
the other.  (from GIT)

--- src/XGMotion.c.orig	2007-09-05 12:41:58.000000000 -0400
+++ src/XGMotion.c
@@ -115,6 +115,8 @@ Time stop;
     savp = readp = (int *)Xmalloc(size);
     bufp = (int *)Xmalloc(size2);
     if (!bufp || !savp) {
+	Xfree(bufp);
+	Xfree(savp);
 	*nEvents = 0;
 	_XEatData(dpy, (unsigned long)size);
 	UnlockDisplay(dpy);
